logstash-core/src/main/java/org/logstash/ackedqueue/FsUtil.java

```diff
@@ -0,0 +1,45 @@
+package org.logstash.ackedqueue;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * File System Utility Methods.
+ */
+public final class FsUtil {
+
+    private FsUtil() {
+    }
+
+    /**
+     * Checks if the request number of bytes of free disk space are available under the given
+     * path.
+     * @param path Directory to check
+     * @param size Bytes of free space requested
+     * @return True iff the
+     * @throws IOException on failure to determine free space for given path's partition
+     */
+    public static boolean hasFreeSpace(final String path, final long size) throws IOException {
+        final File[] partitions = File.listRoots();
+        File location = new File(path).getCanonicalFile();
+        boolean found = false;
+        while (!found) {
+            if (location == null) {
+                throw new IOException(
+                    String.format("Unable to determine the partition that contains '%s'.", path)
+                );
+            }
+            for (final File partition : partitions) {
+                if (partition.equals(location)) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                location = location.getParentFile();
+            }
+        }
+        return location.getFreeSpace() >= size;
```

Comment:
mainly style/readability comment:
since `location`  cannot be `null` upon entering the `while` loop, why not check for `if (location == null)` just after it is mutated? 

Also, I assume that `location.getParentFile()` returns `null` when trying to call it on the root `location`? 
I'd like to see these corner cases tested...